code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -426,41 +488,46 @@
 	max_ammo = 20
 	multiple_sprites = 1
 
-/obj/item/ammo_magazine/m545/ext
-	name = "extended magazine (5.45mm)"
-	matter = list(MAT_STEEL = 2700)
-	max_ammo = 30
-
-/obj/item/ammo_magazine/m545/empty
-	initial_ammo = 0
-
-/obj/item/ammo_magazine/m545/ext/empty
-	initial_ammo = 0
-
 /obj/item/ammo_magazine/m545/practice
 	name = "magazine (5.45mm practice)"
 	ammo_type = /obj/item/ammo_casing/a545/practice
 
-/obj/item/ammo_magazine/m545/practice/ext
-	name = "extended magazine (5.45mm practice)"
-	max_ammo = 30
-
 /obj/item/ammo_magazine/m545/ap
 	name = "magazine (5.45mm armor-piercing)"
 	ammo_type = /obj/item/ammo_casing/a545/ap
 
-/obj/item/ammo_magazine/m545/ap/ext
-	name = "extended magazine (5.45mm armor-piercing)"
-	max_ammo = 30
+/obj/item/ammo_magazine/m545/bb
+	name = "magazine (5.45mm BB)"
+	ammo_type = /obj/item/ammo_casing/a545/bb
 
 /obj/item/ammo_magazine/m545/hunter
 	name = "magazine (5.45mm hunting)"
 	ammo_type = /obj/item/ammo_casing/a545/hunter
 
-/obj/item/ammo_magazine/m545/hunter/ext
-	name = "extended magazine (5.45mm hunting)"
+/obj/item/ammo_magazine/m545/empty
+	initial_ammo = 0
+
+/obj/item/ammo_magazine/m545/ext
+	name = "extended magazine (5.45mm)"
+	matter = list(MAT_STEEL = 2700)
 	max_ammo = 30
 
+/obj/item/ammo_magazine/m545/ext/bb
+	name = "extended magazine (5.45mm BB)"
+	ammo_type = /obj/item/ammo_casing/a545/bb
+
+/obj/item/ammo_magazine/m545/ext/practice
+	name = "extended magazine (5.45mm practice)"
+
+/obj/item/ammo_magazine/m545/ext/ap
+	name = "extended magazine (5.45mm armor-piercing)"
+
+/obj/item/ammo_magazine/m545/ext/hunter
+	name = "extended magazine (5.45mm hunting)"
```

Comment:
reversing this path chunk is of questionable value at best, but -

missing casing types in most swapped paths.